NotesApp/Reusable/CustomAlertView.swift

```diff
@@ -0,0 +1,130 @@
+    var onDelete: (() -> Void)?
+    
+    private lazy var containerView: UIView = {
+        let view = UIView()
+        view.backgroundColor = UIColor(hex: "#252525")
+        view.layer.cornerRadius = 12
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    private lazy var infoImageView: UIImageView = {
+        let imageView = UIImageView()
+        let symbolConfig = UIImage.SymbolConfiguration(pointSize: 50, weight: .regular)
+        imageView.image = UIImage(systemName: "info.circle.fill", withConfiguration: symbolConfig)
+        imageView.tintColor = UIColor(hex: "#606060")
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    private lazy var messageLabel: UILabel = {
+        let label = UILabel()
+        label.text = "Are you sure you want to\ndelete this note?"
+        label.font = UIFont(name: "Nunito-Regular", size: 20)
+        label.textColor = UIColor(hex: "#CFCFCF")
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    private lazy var cancelButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.setTitle("Cancel", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.backgroundColor = UIColor(hex: "#30BE71")
+        button.layer.cornerRadius = 6
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addAction(UIAction(handler: { [weak self] _ in
+            self?.cancelButtonTapped()
+        }), for: .touchUpInside)
+        return button
+    }()
+    
+    private lazy var deleteButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.setTitle("Delete", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.backgroundColor = UIColor(hex: "#FF0000")
+        button.layer.cornerRadius = 6
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addAction(UIAction(handler: { [weak self] _ in
+            self?.deleteButtonTapped()
+        }), for: .touchUpInside)
+        return button
+    }()
+    
+    // MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup View
+    private func setupView() {
+        backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        setupLayout()
+    }
+    
+    private func setupLayout() {
+        addSubview(containerView)
+        containerView.addSubview(infoImageView)
+        containerView.addSubview(messageLabel)
+        containerView.addSubview(cancelButton)
+        containerView.addSubview(deleteButton)
```

Comment:
Its not the for the layout, create separate func called for example setUpHierarchy() for this